source/_patterns/01-elements/input/input.scss

```diff
@@ -1,50 +1,160 @@
+			)};
+
+		input {
+			padding-inline-end: var(--elm-input-padding-inline-end);
+		}
+	}
 
-	padding: $db-spacing-fixed-regular-xs $db-spacing-fixed-regular-m;
+	&[data-variant][data-icon-after] {
+		--elm-input-padding-inline-end: #{calc(
+				2 * #{$db-spacing-fixed-sm} + 0.425rem + 0.75rem + #{$db-spacing-fixed-xs}
+			)};
 
+		input {
+			padding-inline-end: var(--elm-input-padding-inline-end);
+		}
+
+		.elm-state-icon::after {
+			margin-inline-end: calc(0.75rem + $db-spacing-fixed-xs);
+		}
+	}
+
+	&::before,
+	&::after {
+		--icon-font-family: var(--db-base-icon-font-family);
+		--icon-font-size: var(--db-base-icon-font-size);
+
+		@extend %db-body-sm;
+
+		height: var(--db-sizing-md);
+		display: flex;
+		align-items: center;
+		position: absolute;
+		top: 0;
+		padding: 0 $db-spacing-fixed-xs;
+	}
+
+	&::after {
+		right: 0;
+	}
+
+	// currently not supported by firefox
+	&:has(> input:disabled) {
+		&::before,
+		&::after,
+		.elm-state-icon::after {
+			opacity: 0.5;
+		}
+	}
+}
+
+.elm-input input {
+	@extend %db-bg-neutral-transparent-semi;
+
+	border: 1px solid transparent;
+	border-radius: 4px;
+	height: $db-sizing-md;
+	max-width: 100%;
 	width: 100%;
+	padding: $db-spacing-fixed-xs var(--elm-input-padding-inline-end)
+		$db-spacing-fixed-xs var(--elm-input-padding-inline-start);
 
-	&:not(:placeholder-shown) {
-		padding-bottom: 0;
-		padding-top: to-rem($pxValue: 16);
+	& + label {
+		position: absolute;
+		display: flex;
+		align-items: center;
+		font-size: var(--db-type-body-font-size-md);
+		color: $db-colors-secondary-enabled;
+
+		transform-origin: left;
+		transform: translate(
+			var(--elm-input-padding-inline-start),
+			calc(-1 * var(--db-sizing-md))
+		);
+		transition: transform 0.2s $db-transition-functional-timing;
+
+		height: $db-sizing-md;
+		pointer-events: none;
 	}
 
-	// * TODO: to get evaluated
-	// display: inline-flex;
-	// align-items: center;
-	// justify-content: flex-start;
-	// max-width: 100%;
+	& ~ .description {
+		@extend %db-body-xs;
+		margin: $db-spacing-fixed-2xs $db-spacing-fixed-xs 0;
+
+		display: flex;
+		align-items: flex-start;
+		justify-content: space-between;
+	}
 
 	// TODO: Evaluate whether those could get moved to ../_form-elements.scss
 	&:focus,
-	&:hover {
+	&:not(:placeholder-shown) {
 		&:not(:disabled) {
-			--formElement---borderColor: #{$db-colors-secondary-enabled};
+			padding-bottom: calc(
+				$db-spacing-fixed-xs - var(--db-input-focus-opacity) *
+					$db-spacing-fixed-xs
+			);
+			padding-top: $db-spacing-fixed-xs;
+
+			+ label {
+				@include label-focus-animation();
+			}
 		}
 	}
 
 	// TODO: Evaluate whether those could get moved to ../_form-elements.scss
 	&:disabled,
 	fieldset:disabled & {
 		&,
-		& + .elm-label {
+		& + label {
 			// Decided against cursor: not-allowed, compare to e.g. https://phabricator.wikimedia.org/T121960
-			opacity: 0.4;
+			opacity: 0.25;
+		}
+
+		& ~ .description {
+			opacity: 0;
 		}
```

Comment:
I'm not sure, whether I've asked this before, but why would we want to visually hide the description for deactivated fields?

Reply:
removed that. Could be some inherited and not questionated property ;)